### ISSUE TITLE
Source generator is not aware of binding expressions

### ISSUE THREAD

**Issue by DavidBoike** [Particular/NServiceBus.AzureFunctions.Worker.ServiceBus#151]
It's possible to configure the endpoint name using this:

```
[assembly:NServiceBusTriggerFunction("%ENDPOINT_NAME%")]
```

But the [source generator](https://github.com/Particular/NServiceBus.AzureFunctions.Worker.ServiceBus/blob/master/src/NServiceBus.AzureFunctions.Worker.SourceGenerator/TriggerFunctionGenerator.cs) doesn't know this, so it creates a function name `NServiceBusFunctionEndpointTrigger-%ENDPOINT_NAME%` which is not correct.

Specifying everything manually does work:

```
[assembly:NServiceBusTriggerFunction("%ENDPOINT_NAME%", TriggerFunctionName = “MyFunctionName”)]
```

The source generator should be aware of these issues and either Just Work® or provide an error that if parameter binding is used that the second parameter needs to be provided as well.

Also related, the documentation aspects in https://github.com/Particular/docs.particular.net/issues/5740

**Comment by timbussmann** [comment]
it doesn't look like the `FunctionName` attribute supports binding expressions. When using binding expressions, it's also important to use `%ENDPOINT_NAME%` as the setting as any other value doesn't really work the right way. When creating the endpoint configuration, the configuration is currently hard-coded to look for `%ENDPOINT_NAME%`  so other values will cause the endpoint name to be incorrect and different from the queue name (which correctly resolves the binding).

**Comment by kbaley** [comment]
This will be fixed in version 4.1
